File: 81. Search in Rotated Sorted Array II.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: bool
        """
        temp = nums[0]
        if target == temp:
            return True
        elif target > temp:
            for i in xrange(len(nums)):
                if nums[i] < temp:
                    return False
                if nums[i] == target:
                    return True
            return False
        else:
            for i in xrange(len(nums)-1, -1, -1):
                if nums[i] > temp:
                    return False
                if nums[i] == target:
                    return True
            return False

File: 81. Search in Rotated Sorted Array II.cpp
class Solution {
public:
    bool search(vector<int>& nums, int target) {
        int temp = nums[0];
        if (target == temp) return true;
        else if (target > temp)
        {
            for (int i = 0; i < nums.size(); i++)
            {
                if (nums[i] < temp) return false;
                if (target == nums[i]) return true;
            }
        }
        else
        {
            for (int i = nums.size()-1; i >= 0; i--)
            {
                if (nums[i] > temp) return false;
                if (target == nums[i]) return true;
            }
        }
        return false;
    }
};
